src/core/namespace.c

```diff
@@ -287,10 +288,18 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                // mount() always follow symlinks, so we follow symlinks during the stat() as well
```

Comment:
would be great if the kernel mount() syscall would have a mount flag MS_NOSYMLINK or so, which would disable following of symlinks. Kinda like O_NOFOLLOW, but wouldn't even follow symlinks in the non-final parts of the path...